src/stack/Stack.java

```diff
@@ -27,30 +32,36 @@ public void push(int value) {
 	 * @return
 	 */
 	public int pop() {
-
-		return 0;
+		if (tos < 0) {
+			return 0;
+		} else
+			return data[tos--];
 	}
 
 
 	/**
 	 * Clear stack
 	 */
 	public void clear() {
-
+		for (int i = 0; i < DEFAULT_SIZE; i++) {
+			data[i] = 0;
+		}
 	}
 
 	/**
 	 * Check if stack is empty
 	 * @return
 	 */
 	public boolean isEmpty() {
-		return false;
+		return return DEFAULT_SIZE == 0;
 	}
 
 	/**
 	 * Increase the stack capacity if there is not enough space to add additional items
 	 */
 	private void ensureCapacity() {
```

Comment:
add new bigger array and copy the old array there, after it, must use new array as a current